File: IKMS-client/ngapp/src/app/adminSiteModule/model/user/registration-dto.ts
import { UserRegistrationDTO } from "./userRegistrationDTO";
import { PersonalData } from "../personalData/personal-data";
import { Address } from "../address/address";
import { Employee } from "../employee/employee";
import { Parent } from "../../menu/model/parent/parent";

export class RegistrationDto {
    userRegistrationDto: UserRegistrationDTO;
    personalData: PersonalData;
    addressess: Address[];
    employee: Employee;
    parent: Parent;
    
    constructor() {
    }
}

File: IKMS-server/src/main/java/pl/politechnika/ikms/rest/dto/user/RegistrationDto.java
package pl.politechnika.ikms.rest.dto.user;

import lombok.Data;
import pl.politechnika.ikms.rest.dto.person.AddressDto;
import pl.politechnika.ikms.rest.dto.person.EmployeeDto;
import pl.politechnika.ikms.rest.dto.person.ParentDto;
import pl.politechnika.ikms.rest.dto.person.PersonalDataDto;

import java.util.List;

@Data
public class RegistrationDto {

    private UserRegistrationDto userRegistrationDto;

    private PersonalDataDto personalData;

    private List<AddressDto> addressess;

    private EmployeeDto employee;

    private ParentDto parent;
}
